### ISSUE TITLE
[BUG] Erroneous violation on label[for] used for radio button. 

### ISSUE THREAD

**Issue by ConduciveMammal** [IBMa/equal-access#220]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [x] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

> Node version:
> Platform:  Chrome 85.0.4183.102 - macOS Catalina

When working on a site, I've noticed a consistent "Violation" within DAP. My accessibility knowledge isn't 100% so perhaps I'm misunderstanding something.

The error I get is:
>The <label> ‘for’ attribute value of myRadioOption is not the ‘id’ of a valid `<input>` element

When using the markup:

```html
<div class="product-option__swatch">
  <label for="myRadioOption">
    <img src="... 1x" srcset="... 2x" alt="Image" class="image">
  </label>
  <input type="radio" id="myRadioOption" value="Ash Collection" checked hidden>
</div>
```

Is this correct behaviour? If so, what's the cause?

**Comment by tombrunet** [comment]
This is working as expected. Your visible label references a hidden input. Note that this can be a problem because you can click the label and push focus to a hidden location.

If you fix that, you'll get another error because labels for radio buttons are expected to be after the radio button.
